nodes/node.go

```diff
@@ -41,10 +42,15 @@ type Node interface {
 	Init(*types.NodeConfig, ...NodeOption) error
 	Config() *types.NodeConfig
 	PreDeploy(configName, labCADir, labCARoot string) error
-	Deploy(context.Context, runtime.ContainerRuntime) error
-	PostDeploy(context.Context, runtime.ContainerRuntime, map[string]Node) error
+	Deploy(context.Context) error
+	PostDeploy(context.Context, map[string]Node) error
 	WithMgmtNet(*types.MgmtNet)
-	SaveConfig(context.Context, runtime.ContainerRuntime) error
+	WithRuntime(runtime.ContainerRuntime)
+	SaveConfig(context.Context) error
+	Delete(context.Context) error
+	GetImages() []string
```

Comment:
why do you think the order is important? We still build the `map[string]string` inside the `VerifyImages` before we pull them, so that should take care of any duplicates.

Reply:
I mean order is not important, but seeing a `[]string` implies that the order is important.
The idea data struct here is a set not a list, since that doesn't exist in Go, the next best thing is a map.